src/formatDate.js

```diff
@@ -7,8 +7,39 @@
  *
  * @returns {string}
  */
+
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  // Розділяємо дату на частини за допомогою роздільника у форматі fromFormat
+  const separator = fromFormat.pop(); // Останній елемент - це роздільник
+  const dateParts = date.split(separator);
+
+  // Створюємо об'єкт, що містить частини дати відповідно до fromFormat
+  const dateObj = {};
+
+  fromFormat.forEach((part, i) => {
+    dateObj[part] = dateParts[i];
+  });
+
+  // Конвертація формату року з YY в YYYY або назад
+  if (dateObj.YYYY === undefined && dateObj.YY) {
+    const year = Number(dateObj.YY);
+    // Додаємо перевірку для року '00'
+
+    if (year === 0) {
+      dateObj.YYYY = '2000';
+    } else {
+      dateObj.YYYY =
+        year < 30 ? '20' + String(year).padStart(2, '0') : '19' + year;
+    }
+  } else if (dateObj.YY === undefined && dateObj.YYYY) {
+    dateObj.YY = dateObj.YYYY.slice(-2);
+  }
+
+  // Формуємо нову дату у форматі toFormat
+  const newSeparator = toFormat.pop();
+  const newDate = toFormat.map((part) => dateObj[part]).join(newSeparator);
```

Comment:
When mapping over `toFormat` to create the new date string, there is a possibility that `dateObj[part]` may not exist if the `part` is not present in the `fromFormat`. This would result in `undefined` being joined into the string. You should ensure that all parts in `toFormat` are available in `dateObj`.